tools/helpers.cmake

```diff
@@ -553,13 +553,22 @@ function(add_config_library prefix configure_template)
     execute_process(
         COMMAND
             "${PYTHON3}" "${CONFIG_GEN_PATH}" "${config_yaml_file}" --write-c
-            "${config_header_file}" --write-json "${config_json_file}"
+            "${config_header_file}.temp" --write-json "${config_json_file}.temp"
         RESULT_VARIABLE error
     )
     if(error)
         message(FATAL_ERROR "Failed to generate header: ${config_yaml_file}")
     endif()
 
+    execute_process(
+        COMMAND
+            ${CMAKE_COMMAND} -E copy_if_different "${config_header_file}.temp"
+            "${config_header_file}"
+        COMMAND
+            ${CMAKE_COMMAND} -E copy_if_different "${config_json_file}.temp" "${config_json_file}"
+    )
+    file(REMOVE "${config_header_file}.temp" "${config_json_file}.temp")
+
```

Comment:
I'd add a short comment before the new execute_process to say what the purpose of that code is e.g. "Avoid rebuilding everything if nothing changed after reconfiguration".